test/include/seqan3/test/pretty_printing.hpp

```diff
@@ -22,7 +22,8 @@ namespace seqan3
 //!\cond DEV
 //!\brief Overload for the googletest PrintTo function that always delegates to our debug_stream.
 template <typename t>
-    requires true // tricks the compiler to consider this as more specialized than googletests generic PrintTo
+    requires (!std::input_or_output_iterator<t>)
```

Comment:
why do you need this?

Reply:
I wouldn't add this if we don't need this.

See https://github.com/seqan/seqan3/pull/1764/commits/ea2d5d29788959fa6340d3e20adbeebef3e6e218 for an explanation.

Reply:
I didn't suspect that you added this without a reason, I was just asking why because I didn't see where you would like to print iterators. Maybe we should change this to "requires debug-streamable" instead?

Reply:
We are not able to determine in the current design whether a type is debug-streamable or not.

debug_stream says always "I can", but fails when instantiating. This and some other issues with debug-stream lead to this (incomplete) proposal / issue list https://github.com/seqan/product_backlog/issues/63.